lib/db_connection/connection.ex

```diff
@@ -169,23 +169,25 @@ defmodule DBConnection.Connection do
         mon = Process.monitor(pid)
         handle_checkout({ref, mon}, timeout, from, state, s)
       %{client: :closed} ->
-        err = DBConnection.Error.exception("connection not available")
+        message = "connection not available because of disconnection"
+        err = DBConnection.Error.exception(message)
         {:reply, {:error, err}, s}
       %{queue: queue} when queue? == true ->
         client = {ref, Process.monitor(pid)}
         queue = :queue.in({client, timeout, from}, queue)
         {:noreply, %{s | queue: queue}}
       _ when queue? == false ->
-        err = DBConnection.Error.exception("connection not available")
+        message = "connection not available because of queue"
```

Comment:
Unfortunately it may not be true. There is an race condition when using poolboy where poolboy detects a client process has exited and checks out the connection to a new client and the new client tries a checkout before the connection has realised the old client has exited.